api/src/main/java/jakarta/data/page/Slice.java

```diff
@@ -98,15 +99,15 @@ public interface Slice<T> extends Streamable<T> {
     <E> PageRequest<E> pageRequest(Class<E> entityClass);
 
     /**
-     * Returns a request for the {@link PageRequest#next() next} page, or <code>null</code> if it is known that there is no next page.
+     * Returns a request for the {@link PageRequest#next() next} page, or {@link Optional#empty()} if it is known that there is no next page.
      *
      * @return a request for the next page.
      */
-    PageRequest<T> nextPageRequest();
+    Optional<PageRequest<T>> nextPageRequest();
```

Comment:
@njr-11 it still require this validation.

Reply:
That's only because the other pull missed the part about raising an exception from nextPageRequest/previousPageRequest instead of returning null.  I can make sure that gets included when updating the TCK to cover hasNext/hasPrevious.   Then this will follow the pattern of. java.util.Iterator and java.util.ListIterator where you must check hasNext (or hasPrevious) before requesting the next/previous